src/diffusers/pipelines/flux/pipeline_flux.py

```diff
@@ -868,19 +873,19 @@ def __call__(
         else:
             guidance = None
 
-        # TODO: Clarify this section
         if (ip_adapter_image is not None or ip_adapter_image_embeds is not None) and (
             negative_ip_adapter_image is None and negative_ip_adapter_image_embeds is None
         ):
-            negative_ip_adapter_image = np.zeros((width, height, 3), dtype=np.uint8)
-            if isinstance(ip_adapter_image, list):
-                negative_ip_adapter_image = [negative_ip_adapter_image] * len(ip_adapter_image)
+            zeros_image = np.zeros((width, height, 3), dtype=np.uint8)
+            negative_ip_adapter_image_embeds = self.encode_image(zeros_image, device, 1)[None, :]
+            negative_ip_adapter_image_embeds = [negative_ip_adapter_image_embeds] * self.transformer.encoder_hid_proj.num_ip_adapters
```

Comment:
```suggestion
            negative_ip_adapter_image = np.zeros((width, height, 3), dtype=np.uint8)
            if isinstance(ip_adapter_image, list):
                negative_ip_adapter_image = [negative_ip_adapter_image] * self.transformer.encoder_hid_proj.num_ip_adapters
```

Let's keep it consistent with other pipelines. `num_ip_adapters` works as a multiplier. Technically inferring once then repeating is faster but negligible.